### PR TITLE
Add relative path mechanism for dumping

### PR DESCRIPTION
**Pull request by alejoe91**
+ added some docstrings in related functions

### CODE REVIEW

File: spikeinterface/core/base.py [hunk 1]
```diff
@@ -526,10 +560,23 @@ def save_to_folder(self, name=None, folder=None, dump_ext='json', verbose=True,
...
+    dcopy = deepcopy(d)
+    if "kwargs" in dcopy.keys():
+        relative_kwargs = _make_paths_relative(dcopy["kwargs"], relative)
+        dcopy["kwargs"] = relative_kwargs
+        return dcopy
+    else:
+        for k in d.keys():
+            # in SI, all input paths have the "path" keyword
+            if "path" in k:
+                d[k] = str(Path(d[k]).relative_to(relative))
```
Comment (samuelgarcia) [inline spikeinterface/core/base.py:578]: OK I see. The error is here.
path can be str/Path or list of str/Path. You have to take this in account.
Please make a test with a multi segment object.

Comment (alejoe91) [reply spikeinterface/core/base.py:578]: Can you point me to an existing test with multi segment objects?

Comment (samuelgarcia) [reply spikeinterface/core/base.py:578]: https://github.com/SpikeInterface/spikeinterface/blob/master/spikeinterface/core/tests/testing_tools.py#L7

File: spikeinterface/core/base.py [hunk 2]
```diff
@@ -275,20 +275,24 @@ def to_dict(self, include_annotations=False, include_properties=False, include_f
         return dump_dict
 
     @staticmethod
-    def from_dict(d):
+    def from_dict(d, base_folder=None):
```
Comment (samuelgarcia) [inline spikeinterface/core/base.py:278]: This is OK.

File: spikeinterface/core/base.py [hunk 3]
```diff
@@ -422,7 +426,7 @@ def load(file_path):
                     d = pickle.load(f)
             else:
                 raise ValueError(f'Impossible to load {file_path}')
-            extractor = BaseExtractor.from_dict(d)
+            extractor = BaseExtractor.from_dict(d, file_path.parent)
```
Comment (samuelgarcia) [inline spikeinterface/core/base.py:429]: Here I think we need something to check if it is abslute or not.
We need to be flexible to handle both relative and absolite.

Comment (alejoe91) [reply spikeinterface/core/base.py:429]: It checks with exists

**Comment by samuelgarcia** [comment]
I tried some fix. Lets see if it works.

**Review by samuelgarcia — APPROVED** [review]

**Pull request merged**